packages/compartment-mapper/src/parse-cjs-shared-export-wrapper.js

```diff
@@ -8,6 +8,47 @@ const { freeze, keys, create, hasOwnProperty } = Object;
  * @returns {boolean}
  */
 const has = (object, key) => apply(hasOwnProperty, object, [key]);
+const removeProtocol = url => {
+  if (url.substring(0, 7) === 'file://') {
+    url = url.substring(7);
+  }
+  return url;
+};
+const noTrailingSlash = path => path.replace(/[\\/]$/, '');
+
+export const getModulePaths = async (readPowers, url) => {
+  if (!(readPowers && readPowers.fileURLToPath)) {
+    return {
+      filename: null,
+      dirname: null,
+    };
+  }
+
+  let urlFilename = url;
+  let urlDirname;
+  try {
+    urlDirname = new URL('./', url).href;
+  } catch (_) {
+    return {
+      filename: null,
+      dirname: null,
+    };
+  }
+
+  if (readPowers && readPowers.fileURLToPath) {
+    urlFilename = readPowers.fileURLToPath(urlFilename).toString();
+    urlDirname = readPowers.fileURLToPath(urlDirname).toString();
+  } else {
+    urlFilename = removeProtocol(urlFilename);
+    urlDirname = removeProtocol(urlDirname);
+  }
+  urlDirname = noTrailingSlash(urlDirname);
+
+  return {
+    filename: urlFilename,
+    dirname: urlDirname,
```

Comment:
Slightly more concise with just `filename` instead of `urlFilename`. I do not feel strongly.